libexec/tfenv-install

```diff
@@ -105,9 +109,35 @@ if [[ -n "${keybase_bin}" && -x "${keybase_bin}" ]]; then
     warn_and_continue "Unable to verify GPG signature unless logged into keybase and following hashicorp"
   else
     info "Downloading SHA hash signature file from ${version_url}/${shasums_name}.sig"
-    curlw -s -f -o "${download_tmp}/${shasums_name}.sig" "${version_url}/${shasums_name}.sig" || error_and_die "SHA256SUMS signature download failed"
     "${keybase_bin}" pgp verify -S hashicorp -d "${download_tmp}/${shasums_name}.sig" -i "${download_tmp}/${shasums_name}" || error_and_die "SHA256SUMS signature does not match!"
   fi
+elif [[ $(gpg --version |grep -c RSA ) -gt 1 || $skip_verify -eq 1 ]] ; then
+  info "attempting to verify using GPG"
+  if [[ $UID -eq 0 ]]; then
+    warn_and_continue "Running as root! keybase would have failed, reconsider using a non-elevated user"
+  fi
+
+  # fairly obtuse use of GPG, 12th column is capabilites of the key, capitals are primary key, S is signature
+  if [[ $(gpg --with-colons -k 'security@hashicorp.com' |awk -F: '{print $12}') =~ 'S' ]]; then
+    blind_trust_keyring=""
+  else
+    notice_and_continue "Could not detect hashicorp's public key, continuing; this will not modify the trust store"
+    blind_trust_keyring="--no-default-keyring --keyring /tmp/tfenv_keyring.gpg"
+
+    gpg ${blind_trust_keyring} --batch --import <(curl -s https://keybase.io/hashicorp/pgp_keys.asc)
+    if [[ $? -ne 0 ]]; then
+      gpg ${blind_trust_keyring} --keyserver hkps://hkps.pool.sks-keyservers.net:443 \
+      --recv-keys $(gpg --batch --list-packets "${version_url}/${shasums_name}.sig" | egrep -o ' [1-9A-F]{16}$')
+    fi
+    notice_and_continue "Consider importing the key security@hashicorp.com and trusting it"
+  fi
+  # GPG trusts can be complex, it would be nice to handle the relationship better, for now output to console
+  gpg ${blind_trust_keyring} --batch --verify "${download_tmp}/${shasums_name}.sig" \
+   "${download_tmp}/${shasums_name}" 
+  if [[ $? -ne 0 ]]; then
+    info "Signature has been verified against a signature from Hashicorp"
+  fi
+  rm /tmp/tfenv_keyring.gpg
```

Comment:
There's a temp working dir for the downloads, safely owned by the current user, cleaned up on exit already, so there's no reason to introduce races in `/tmp` and possible security issues that way.